service/accounts/openstack_manager.py

```diff
@@ -485,33 +435,124 @@ def delete_security_group(self, identity):
             self.network_manager.neutron.delete_security_group(sec_group["id"])
         return True
 
-    def delete_network(self, identity, remove_network=True):
-        # Core credentials need to be converted to openstack names
+    def select_network_strategy(self, identity, topology_name=None):
+        """
+        Select a network topology and initialize it with the identity/provider specific information required.
+        """
+        # Select Cls
+        if not topology_name:
+            NetworkTopologyStrategyCls = ExternalRouter
+        else:
+            NetworkTopologyStrategyCls = get_topology_cls(topology_name)
+
+        try:
+            network_strategy = NetworkTopologyStrategyCls(identity)
+        except:
+            raise
+            raise Exception(
+                "Error initializing Network Topology - %s + %s " %
+                (NetworkTopologyStrategyCls, identity))
+        return network_strategy
+
+    def dns_nameservers_for(self, identity):
+        dns_nameservers = [
+            dns_server.ip_address for dns_server
+            in identity.provider.dns_server_ips.order_by('order')
+        ]
+        return dns_nameservers
+
+    def delete_user_network(self, identity, options={}):
+        """
+        1. Look at the provider for network topology hints
+        2. If no network topology exists, use the "Default network" settings.
+        3. Delete network based on topology
+        """
+
         identity_creds = self.parse_identity(identity)
-        username = identity_creds["username"]
         project_name = identity_creds["tenant_name"]
-        # Convert from libcloud names to openstack client names
-        return self.network_manager.delete_project_network(
-            username, project_name, remove_network=remove_network)
+        neutron = self.get_openstack_client(identity, 'neutron')
+        try:
+            topology_name = self.cloud_config['network']['topology']
+        except Exception:
+            logger.exception(
+                "Network topology not selected -- "
+                "Will attempt to use the last known default: ExternalRouter.")
+            topology_name = None
+        network_strategy = self.select_network_strategy(identity, topology_name)
```

Comment:
I see. This could be changed to:
```
        topology_name = getattr(self.cloud_config['network'], 'topology', None)

        network_strategy = None
        if not topology_name:
            logger.exception(
                "Network topology not selected -- "
                "Will attempt to use the last known default: ExternalRouter.")
            network_strategy = self.select_network_strategy(identity)
        else:
            network_strategy = self.select_network_strategy(identity, topology=topology_name)
```

It's easier to verify visually that `topology_name=None` will not have a bad impact on `select_network_strategy`.